wpilibc/shared/include/CircularBuffer.inc

```diff
@@ -82,6 +82,61 @@ T CircularBuffer<T>::PopBack() {
 }
 
 /**
+ * Resizes internal buffer to given size.
+ */
+template <class T>
+void CircularBuffer<T>::Resize(size_t size) {
+  if (size > m_data.size()) {
+    // Find end of buffer
+    size_t insertLocation = (m_front + m_length) % m_data.size();
+
+    // If insertion location precedes front of buffer, push front index back
+    if (insertLocation <= m_front) {
+      m_front += size - m_data.size();
+    }
+
+    // Add elements to end of buffer
+    m_data.insert(m_data.begin() + insertLocation, size - m_data.size(), 0);
+  } else if (size < m_data.size()) {
+    /* 1) Shift element block start at "front" left as many blocks as were
+     *    removed up to but not exceeding buffer[0]
+     * 2) Shrink buffer, which will remove even more elements automatically if
+     *    necessary
+     */
+    size_t elemsToRemove = m_data.size() - size;
+    auto frontIter = m_data.begin() + m_front;
+    if (m_front < elemsToRemove) {
+      // Shift start of element block to left
+      m_data.erase(m_data.begin(), frontIter);
+      elemsToRemove -= m_front;
+
+      // Update metadata
+      m_front = 0;
+
+      // More elements must be removed, so do so from end of buffer
+      m_data.erase(m_data.begin() + m_data.size() - elemsToRemove,
+                   m_data.end());
```

Comment:
Shouldn't the end be erased first so you aren't copying elements with the first erase, just to erase them later?